docs/tools/config_system.md

```diff
@@ -1,23 +1,135 @@
 
-<span class="notes">**Note:** In prior releases, the configuration system provided a method for adding custom targets. The Mbed OS tools now look for custom targets in a file named `custom_targets.json` in the root of a project and treat custom targets the same as [Mbed targets](/docs/development/tools/adding-and-configuring-targets.html).</span>
+<span class="notes">**Note:** In prior releases, the configuration system provided a method for adding custom targets. The Mbed OS tools now look for custom targets in a file named `custom_targets.json` in the root of an application and treat custom targets the same as [Mbed targets](/docs/development/tools/adding-and-configuring-targets.html).</span>
+
+### Examining available configuration parameters
+
+Mbed CLI includes a command for listing and explaining the compiliation time configuration, `mbed compile --config`. This command prints a summary of configuration parameters, such as:
+```
+Configuration parameters
+------------------------
+cellular.random_max_start_delay = 0 (macro name: "MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY")
+cellular.use-apn-lookup = 1 (macro name: "MBED_CONF_CELLULAR_USE_APN_LOOKUP")
+configuration-store.storage_disable = 0 (macro name: "CFSTORE_STORAGE_DISABLE")
+drivers.uart-serial-rxbuf-size = 256 (macro name: "MBED_CONF_DRIVERS_UART_SERIAL_RXBUF_SIZE")
+drivers.uart-serial-txbuf-size = 256 (macro name: "MBED_CONF_DRIVERS_UART_SERIAL_TXBUF_SIZE")
+events.present = 1 (macro name: "MBED_CONF_EVENTS_PRESENT")
+events.shared-dispatch-from-application = 0 (macro name: "MBED_CONF_EVENTS_SHARED_DISPATCH_FROM_APPLICATION")
+events.shared-eventsize = 256 (macro name: "MBED_CONF_EVENTS_SHARED_EVENTSIZE")
+events.shared-highprio-eventsize = 256 (macro name: "MBED_CONF_EVENTS_SHARED_HIGHPRIO_EVENTSIZE")
+events.shared-highprio-stacksize = 1024 (macro name: "MBED_CONF_EVENTS_SHARED_HIGHPRIO_STACKSIZE")
+events.shared-stacksize = 1024 (macro name: "MBED_CONF_EVENTS_SHARED_STACKSIZE")
+<output truncated for brevity>
+```
+
+You may find that configuration naming does not always describe the parameter's purpose adequately. Include the `-v` switch to include the help text defined with the configuration parameter, where the value of the configuration parameter is defined, and other details. The command `mbed compile --config -v` in the same application as above prints:
+```
+Configuration parameters
+------------------------
+Name: cellular.random_max_start_delay
+    Description: Maximum random delay value used in start-up sequence in milliseconds
+    Defined by: library:cellular
+    No value set
+Name: cellular.use-apn-lookup
```

Comment:
Tangent: I see that in cases where the macro name is not defined in the `.json` file, it is not in the verbose listing, but it still is in the brief listing. Is this desirable behavior? Consider adding "Macro name" for all, with "` (default)`" following those that aren't explicitly specified in the `.json` files.

Reply:
I'm wondering how you define a configuration parameter without giving it a name. Could you explain?

Reply:
To be clear, I'm suggesting a change to the tool output in verbose mode, not just the documentation. (Then the documentation here would reflect the new output.)

Example: Since no `"macro_name"` is specified for `"cellular.random_max_start_delay"`, add a line in its verbose output, `Macro name: MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY  (default)`.

Reply:
Odd. Check a macro with a value. It has the behavior you're requesting. I think the fact that it does not show for config params without a value is a bug.

Reply:
That's what I was saying, though I don't know enough about the intent to say whether it's a "bug" per se.

Reply:
It also prints in the short version. Definitely seems like a bug to me.